File: python/deck.py
import random
from card import Card


SUITS = [1,2,3,4]

RANKS = [2,3,4,5,6,7,8,9,10,11,12,13,14]


class Deck:
    def __init__(self):
        self.stk = []
        for i in range(4):
            for j in range(13):
                card = Card(SUITS[i],RANKS[j])
                self.stk.append(card)
                
        random.shuffle(self.stk)
        
    def pop_card(self):
        card = self.stk.pop()
        return card

File: logic/deck.js
// import {Card} from "./card.js";

const shuffle = require(__dirname + "/shuffle.js");
const Card = require(__dirname + "/card.js");

let SUITS = ["clubs", "diamonds", "hearts", "spades"];

let RANKS = ["2","3","4","5","6","7","8","9","10","jack","queen","king","ace"];


class Deck {
    constructor() {
        this.stck = [];
        this.value = 12;
        for (let i = 0; i < 4; i++) {
            for (let j = 0; j < 13; j++) {
                let card = new Card(SUITS[i], RANKS[j]);
                this.stck.push(card);
            }
        }
    }
    printCard() {
        for (let i = 0; i < this.stck.length; i++) {
            console.log(this.stck[i].filename);
        }
    }
    popCard() {
        let card = this.stck.pop();
        return card;
    }
    shuffleCards() {
        this.stck = shuffle(this.stck)
    }
}



module.exports = Deck;
// let deck = new Deck();

// deck.printCard();